src/types/custom.ts

```diff
@@ -0,0 +1,20 @@
+import { Runtype, Rt, Static, create, validationError } from '../runtype'
+import { String } from './string'
+
+export interface Custom<A extends Rt, T extends string, K> extends Runtype<Static<A>> {
+  tag: T,
+  underlying: A,
+  args: K
+}
+
+export function Custom<A extends Rt, T extends string, K>(underlying: A, constraint: (x: Static<A>, args: K) => boolean | string, tag: T, args: K) {
```

Comment:
Why does the `constraint` argument need to be

```ts
constraint: (x: Static<A>, args: K) => boolean | string
```

instead of just

```ts
constraint: (x: Static<A>) => boolean | string
```

You're already defining `constraint` within the scope of `args`, so you don't need to pass it in, do you?